From 26c59f5ce7fac44150e5986cef33808c79cd9ded Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Wed, 22 Sep 2021 10:05:41 +0200
Subject: [PATCH] Use libxml version instead of PHP version before calling
 libxml_disable_entity_loader

In some cases older version of php 7.4 are using libxml > 2.9.0 so use
the libxml version instad of the php 8.0 to see if the call of libxml is
useful.

Otherwise this creates some warning.

Signed-off-by: Carl Schwan <carl@carlschwan.eu>
---
 lib/Reader.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Reader.php b/lib/Reader.php
index 368e8ff..1a27af7 100644
--- a/lib/Reader.php
+++ b/lib/Reader.php
@@ -56,7 +56,7 @@ public function getClark()
     public function parse(): array
     {
         $previousEntityState = null;
-        $shouldCallLibxmlDisableEntityLoader = (\PHP_VERSION_ID < 80000);
+        $shouldCallLibxmlDisableEntityLoader = (\LIBXML_VERSION < 20900);
         if ($shouldCallLibxmlDisableEntityLoader) {
             $previousEntityState = libxml_disable_entity_loader(true);
         }
